### PR TITLE
SOV-2117: Add stability pool subsidies details to the Earn and Rewards pages

### PR DESCRIPTION
**Pull request by Rickk137**
https://sovryn.atlassian.net/browse/SOV-2117

### CODE REVIEW

File: apps/frontend/src/app/5_pages/EarnPage/EarnPage.tsx [hunk 1]
```diff
@@ -438,6 +441,24 @@ const EarnPage: FC = () => {
             />
           )}
 
+          <SimpleTable className="mt-3">
+            <SimpleTableRow
+              label={
+                <span className="relative">
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/EarnPage/EarnPage.tsx]: `HelperButton` is not vertically centered, use divs and flexbox.

image

File: apps/frontend/src/app/5_pages/EarnPage/EarnPage.tsx [hunk 2]
```diff
@@ -438,6 +441,24 @@ const EarnPage: FC = () => {
...
+                  <HelperButton
+                    className="absolute -right-1 top-0 translate-x-full"
+                    content={t(pageTranslations.subsidiesRewardRateInfo)}
+                  />
+                </span>
+              }
+              valueClassName={classNames('transition-colors', {
+                'text-primary-10': !isAmountZero,
+              })}
+              value={`${formatValue(apy, 5)}% APY`}
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/EarnPage/EarnPage.tsx]: I already mentioned it in the story but the value should be rounded to 2 decimals.

File: apps/frontend/src/app/5_pages/EarnPage/hooks/useGetSubsidiesAPR.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { getContract } from '@sovryn/contracts';
+
+import { useAccount } from '../../../../hooks/useAccount';
+import { getRskChainId } from '../../../../utils/chain';
+
+export const useGetSubsidiesAPR = () => {
+  const { signer } = useAccount();
+  const [apr, setAPR] = useState(0);
+
+  const getCommunityIssuance = useCallback(async () => {
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/EarnPage/hooks/useGetSubsidiesAPR.ts]: Use `useLoadContract` hook instead.

File: apps/frontend/src/app/5_pages/EarnPage/hooks/useGetSubsidiesAPR.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  const updateAPR = useCallback(async () => {
+    const communityIssuance = await getCommunityIssuance();
+    const APR = await communityIssuance.APR();
+    setAPR(Number(APR.toString()) / 100);
+  }, [getCommunityIssuance]);
+
+  const apy = useMemo(() => {
+    return (Math.pow(1 + apr / 36525, 365.25) - 1) * 100;
+  }, [apr]);
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/EarnPage/hooks/useGetSubsidiesAPR.ts]:
```suggestion
  const apy = useMemo(
    () => (Math.pow(1 + apr / 36525, 365.25) - 1) * 100,
    [apr],
  );
```

File: apps/frontend/src/app/5_pages/EarnPage/hooks/useGetSubsidiesAPR.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  useEffect(() => {
+    updateAPR();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/EarnPage/hooks/useGetSubsidiesAPR.ts]: Do you need it?

File: apps/frontend/src/app/5_pages/RewardsPage/hooks/useGetSovGain.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    const { address, abi: stabilityPoolAbi } = await getContract(
+      'stabilityPool',
+      'zero',
+      getRskChainId(),
+    );
+    const stabilityPool = new ethers.Contract(
+      address,
+      stabilityPoolAbi,
+      signer,
+    );
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/RewardsPage/hooks/useGetSovGain.ts]: Use `useLoadContract` hook.

File: apps/frontend/src/app/5_pages/RewardsPage/hooks/useGetSovGain.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+    const gain = await stabilityPool.getDepositorSOVGain(account);
+    setSOVGain(gain.toString());
+  }, [account, signer]);
+
+  useEffect(() => {
+    updateSOVGain();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/RewardsPage/hooks/useGetSovGain.ts]: Is it necessary?

**Review by tiltom — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/5_pages/RewardsPage/hooks/useGetSovGain.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+  const stabilityPool = useLoadContract(
+    'stabilityPool',
+    'zero',
+    getRskChainId(),
+  );
+
+  const updateSOVGain = useCallback(async () => {
+    const gain = await stabilityPool?.getDepositorSOVGain(account);
+    setSOVGain(formatUnits(gain.toString()));
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/RewardsPage/hooks/useGetSovGain.ts]:
```suggestion
    setSOVGain(fromWei(gain.toString()));
```
Use `fromWei` from `utils/math.ts`. It's using `formatUnits` under the hood but I prefer to use our helpers.

File: apps/frontend/src/app/5_pages/RewardsPage/hooks/useGetSovGain.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+export const useGetSovGain = () => {
+  const { account } = useAccount();
+  const [sovGain, setSOVGain] = useState('0');
+
+  const stabilityPool = useLoadContract(
+    'stabilityPool',
+    'zero',
+    getRskChainId(),
+  );
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/RewardsPage/hooks/useGetSovGain.ts]:
```suggestion
  const stabilityPool = useLoadContract(
    'stabilityPool',
    'zero'
  );
```
No need to specify the chain id (unless you want to use something else than RSK).

File: apps/frontend/src/app/5_pages/EarnPage/hooks/useGetSubsidiesAPR.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+import { getRskChainId } from '../../../../utils/chain';
+
+export const useGetSubsidiesAPR = () => {
+  const [apr, setAPR] = useState(0);
+
+  const communityIssuance = useLoadContract(
+    'communityIssuance',
+    'zero',
+    getRskChainId(),
+  );
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/EarnPage/hooks/useGetSubsidiesAPR.ts]:
```suggestion
  const communityIssuance = useLoadContract(
    'communityIssuance',
    'zero'
  );
```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Pull request merged**